### PR TITLE
ELPP-3323 Use single front controller

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Requires https://github.com/elifesciences/journal/pull/954

### CODE REVIEW

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 1]
```diff
@@ -80,12 +80,13 @@ server {
...
     {% if pillar.elife.env == 'dev' %}
-    location /app_test.php {
+    location /app.php {
         fastcgi_split_path_info ^(.+\.php)(/.+)$;
         include fastcgi_params;
-        fastcgi_param SCRIPT_FILENAME /srv/journal/web/app_test.php;
+        fastcgi_param SCRIPT_FILENAME /srv/journal/web/app.php;
         fastcgi_param DOCUMENT_ROOT $realpath_root;
         fastcgi_param ENVIRONMENT_NAME {{ pillar.elife.env }};
+        fastcgi_param APP_ENV {{ pillar.elife.env }};
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: if this is specific to Selenium, should it be hardcoded to `test`?

File: salt/journal/config/etc-nginx-sites-enabled-journal-local-demo.conf [hunk 2]
```diff
@@ -11,15 +11,16 @@ server {
     }
 
     location / {
-        try_files $uri /app_{{ environment_name }}.php$is_args$query_string;
+        try_files $uri /app.php$is_args$query_string;
     }
 
-    location ~ ^/app_{{ environment_name }}\.php(/|$) {
+    location /app.php {
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-enabled-journal-local-demo.conf]: some of these `location` have lost the final `/` but I guess it's unnecessary now that the match is being done with a prefix

File: salt/journal/config/etc-nginx-sites-available-journal.conf (hunk 1 above)
Comment (thewilkybarkid) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: Haven't tried but I think this is broken: needs to be two separate sites. Not sure it's worth fixing, only used inside the VM which is unused and to shortly be replaced?

Comment (thewilkybarkid) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: Though `location /app_test.php` could remain the `APP_ENV` set to `dev`? `behat.yml` needs reverting too in that case.

Comment (giorgiosironi) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: Given we are not running any test in the Vagrant VM and this doesn't influence `journal--ci` (it's a `dev` block), this section can be removed altogether if it makes things easier.

**Pull request merged**